File: src/main/java/raylras/zen/ast/expr/MemberAccessExpression.java
package raylras.zen.ast.expr;

import raylras.zen.ast.ASTVisitor;

public class MemberAccessExpression extends Expression {

    private final Expression left;
    private final Expression right;

    public MemberAccessExpression(Expression left, Expression right) {
        this.left = left;
        this.right = right;
    }

    public Expression getLeft() {
        return left;
    }

    public Expression getRight() {
        return right;
    }

    @Override
    public void accept(ASTVisitor visitor) {
        visitor.visitMemberAccessExpression(this);
        left.accept(visitor);
        right.accept(visitor);
    }

}

File: zenscript-code-model/src/main/kotlin/raylras/zen/model/ast/expr/MemberAccessExpression.kt
package raylras.zen.model.ast.expr

import com.strumenta.kolasu.model.Expression
import com.strumenta.kolasu.model.Named
import com.strumenta.kolasu.model.Node
import com.strumenta.kolasu.model.ReferenceByName

data class MemberAccessExpression(
    val receiver: Expression,
    val member: ReferenceByName<Named>,
) : Node(), Expression
